src/System.Memory/src/System/Buffers/StandardFormat.cs

```diff
@@ -69,33 +71,41 @@ public StandardFormat(char symbol, byte precision = NoPrecision)
         /// </summary>
         public static StandardFormat Parse(ReadOnlySpan<char> format)
         {
-            return Parse(new string(format.ToArray()));
-        }
-
-        /// <summary>
-        /// Converts a classic .NET format string into a StandardFormat
-        /// </summary>
-        public static StandardFormat Parse(string format)
-        {
-            if (string.IsNullOrEmpty(format))
+            if (format.Length == 0)
                 return default;
 
-            char specifier = format[0];
-            byte precision = NoPrecision;
-
-            if (format.Length > 1)
+            char symbol = format[0];
+            byte precision;
+            if (format.Length == 1)
             {
-                if (!byte.TryParse(format.Substring(1), out precision))
-                    throw new FormatException("format");
-
-                if (precision > MaxPrecision)
-                    throw new FormatException("precision");
+                precision = NoPrecision;
+            }
+            else
+            {
+                uint parsedPrecision = 0;
+                for (int srcIndex = 1; srcIndex < format.Length; srcIndex++)
+                {
+                    uint digit = format[srcIndex] - 48u; // '0'
+                    if (digit > 9)
+                        throw new FormatException(SR.Format(SR.Argument_CannotParsePrecision, MaxPrecision));
+
+                    parsedPrecision = parsedPrecision * 10 + digit;
+                    if (parsedPrecision > MaxPrecision)
+                        throw new FormatException(SR.Format(SR.Argument_PrecisionTooLarge, MaxPrecision));
+                }
+
+                precision = (byte)parsedPrecision;
             }
 
-            return new StandardFormat(specifier, precision);
+            return new StandardFormat(symbol, precision);
         }
 
         /// <summary>
+        /// Converts a classic .NET format string into a StandardFormat
+        /// </summary>
+        public static StandardFormat Parse(string format) => format == null ? default : Parse(SpanExtensions.AsReadOnlySpan(format));  //@todo: Change back to extension syntax once the ambiguous reference with CoreLib is eliminated.
```

Comment:
Can you explain why this is causing an ambiguous reference?

Reply:
This:

https://github.com/dotnet/coreclr/blob/6a5b0e5bd45b987005d0947cec35dce9eb2c51f2/src/mscorlib/shared/System/Span.NonGeneric.cs#L121

paired with this:

https://github.com/dotnet/corefx/blob/ea39ed02605c52876d539275af367cd6073b581d/src/System.Memory/src/System.Memory.csproj#L69

Reply:
I do not understand the issue. If there is an ambiguous reference, how will it be eliminated?
 
We reference CoreLib and then the extension methods forward to them.
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/SpanExtensions.Fast.cs#L48

Reply:
One extension method is defined on `SpanExtensions`, the other on `Span`. Presumably, the fix is to fix CoreLib - either by making its own copy of the extension method internal or by moving it to the correct type. It's out of the scope of this PR.

Reply:
Or perhaps the CoreLib `AsReadOnlySpan` shouldn't be an extension method.